simde/x86/sse4.2.h

```diff
@@ -37,6 +37,232 @@ SIMDE__BEGIN_DECLS
 #  define SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES
 #endif
 
+#define SIMDE_SIDD_CMP_EQUAL_ANY     0
+#define SIMDE_SIDD_CMP_RANGES     1
+#define SIMDE_SIDD_CMP_EQUAL_EACH     2
+#define SIMDE_SIDD_CMP_EQUAL_ORDERED   3
+
+#if defined(SIMDE_X86_SSE4_2_ENABLE_NATIVE_ALIASES)
+#define _SIDD_CMP_EQUAL_ANY SIMDE_SIDD_CMP_EQUAL_ANY
+#define _SIDD_CMP_RANGES SIMDE_SIDD_CMP_RANGES
+#define _SIDD_CMP_EQUAL_EACH SIMDE_SIDD_CMP_EQUAL_EACH
+#define _SIDD_CMP_EQUAL_ORDERED SIMDE_SIDD_CMP_EQUAL_ORDERED
+#endif
+
+SIMDE__FUNCTION_ATTRIBUTES
+int
+simde_mm_cmpestra_8_(simde__m128i a, int la, simde__m128i b, int lb, const int imm8) {
+  const int cmp_op = imm8 & 0x06;
+  const int polarity = imm8 & 0x30;
+  simde__m128i_private
+    bool_res_ = simde__m128i_to_private(simde_mm_setzero_si128()),
+    a_ = simde__m128i_to_private(a),
+    b_ = simde__m128i_to_private(b);
+  const int upper_bound = (128 / 8) - 1;
+  int a_invalid = 0;
+  int b_invalid = 0;
+  for (size_t i = 0 ; i < (upper_bound) ; i++) {
```

Comment:
``` cc
simde/x86/sse4.2.h:64:25: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   for (size_t i = 0 ; i < (upper_bound) ; i++) {
                         ^
```
```suggestion
  for (int i = 0 ; i < (upper_bound) ; i++) {
```

Reply:
I have corrected all bugs of this type in new commit.

Reply:
FWIW, the reason I use size_t everywhere in SIMDe is that we generally compare to the result of the `sizeof` operator, which is `size_t`.  Here, however, since `upper_bound` is an `int` it makes sense to use `int`.